src/unstract/adapters/x2text/llm_whisperer/src/llm_whisperer.py

```diff
@@ -320,3 +317,35 @@ def process(
             logger.error(f"OS error while writing {output_file_path}: {e} ")
             raise ExtractorError(str(e))
         return output
+
+    def process(
+        self,
+        input_file_path: str,
+        output_file_path: Optional[str] = None,
+        **kwargs: dict[Any, Any],
+    ) -> dict[str, Any]:
+        """Used to extract text from documents.
+
+        Args:
+            input_file_path (str): Path to file that needs to be extracted
+            output_file_path (Optional[str], optional): File path to write
+                extracted text into, if None doesn't write to a file.
+                Defaults to None.
+
+        Returns:
+            str: Extracted text
+        """
+
+        output = {}
+
+        response: requests.Response = self._send_whisper_request(
+            input_file_path, bool(kwargs.get(X2TextConstants.ENABLE_HIGHLIGHT, False))
+        )
+
+        output[X2TextConstants.EXTRACTED_TEXT] = self._extract_text_from_response(
+            output_file_path, response
+        )
+        output[X2TextConstants.WHISPER_HASH] = response.headers.get(
+            X2TextConstants.WHISPER_HASH, ""
+        )
```

Comment:
@johnyrahul consider wrapping this in a `dataclass` which could have a dict `extraction_metadata` - this could in turn highlight type of extractor and `whisper_hash` could be a field in this. This is so that we can easily parse the output irrespective of the extractor used

Reply:
Addressed can you re review this part.